### PR TITLE
Add new service Impl methods to account for the new Country fields

### PR DESCRIPTION
**Pull request by alee8888**
@drewbrokke @pei-jung

### CODE REVIEW

**Comment by alee8888** [comment]
ci:test:sf

File: portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java [hunk 1]
```diff
@@ -14,11 +14,253 @@
...
+
 /**
  * @author Brian Wing Shun Chan
  * @see CountryLocalServiceBaseImpl
  */
 public class CountryLocalServiceImpl extends CountryLocalServiceBaseImpl {
+
+	@Override
+	public Country addCountry(
+			boolean active, String a2, String a3, boolean billingAllowed,
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java:47]: The parameter order here should match the service.xml file. So `active` should come after `a2` and `a3`.

File: portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java [hunk 2]
```diff
@@ -14,11 +14,253 @@
...
+		country.setA3(a3);
+		country.setActive(active);
+		country.setBillingAllowed(billingAllowed);
+		country.setDefaultLanguageId(defaultLanguageId);
+		country.setIdd(idd);
+		country.setName(name);
+		country.setNumber(number);
+		country.setPosition(position);
+		country.setShippingAllowed(shippingAllowed);
+		country.setSubjectToVAT(subjectToVAT);
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java:83]: I think Commerce also sets the filter field here:

```commerceCountry.setChannelFilterEnabled(false);```

File: portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java [hunk 3]
```diff
@@ -14,11 +14,253 @@
...
+
+		Country country = countryPersistence.findByPrimaryKey(countryId);
+
+		country.setActive(active);
+
+		return countryPersistence.update(country);
+	}
+
+	@Override
+	public Country updateCommerceCountryChannelFilter(
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java:205]: Our service doesn't have a channel concept. This method should be named `updateGroupFilterEnabled`.

File: portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java [hunk 4]
```diff
@@ -14,11 +14,253 @@
...
+		Country country = countryLocalService.getCountry(countryId);
+
+		country.setGroupFilterEnabled(groupFilterEnabled);
+
+		return countryPersistence.update(country);
+	}
+
+	@Override
+	public Country updateCountry(
+			long countryId, boolean active, String a2, String a3,
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java:218]: Same as the `add` method, `active` should come after `a2` and `a3`.

File: portal-impl/src/com/liferay/portal/service/impl/CountryServiceImpl.java [hunk 5]
```diff
@@ -19,20 +19,42 @@
...
+import java.util.Map;
 
 /**
  * @author Brian Wing Shun Chan
  */
 public class CountryServiceImpl extends CountryServiceBaseImpl {
 
+	@Override
+	public Country addCountry(
+			boolean active, String a2, String a3, boolean billingAllowed,
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CountryServiceImpl.java:42]: See my comments in the local service, the same applies to the `add` and `update` methods here.

File: portal-impl/src/com/liferay/portal/service/impl/CountryServiceImpl.java [hunk 6]
```diff
@@ -68,60 +90,237 @@ public Country addCountry(
...
+	@Override
+	public List<Country> getCountriesByCompanyId(
+			long companyId, int start, int end,
+			OrderByComparator<Country> orderByComparator)
+		throws PortalException {
+
+		if (!getPermissionChecker().isOmniadmin()) {
+			throw new PrincipalException.MustBeOmniadmin(
+				getPermissionChecker());
+		}
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CountryServiceImpl.java:188]: It doesn't really make sense that `getCountriesByCompanyId(long companyId)` is not permission checked, but this method is. I think we can just not permission check `get` and `fetch` methods for now. Only permission check the `write` methods.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
Hey @alee8888, there are a few changes still needed for this ticket. Please see my inline comments. We're close!

**Pull request closed without merging**